tools/image_processing/imagej2/imagej2_bunwarpj_align.xml

```diff
@@ -1,75 +1,189 @@
-        --landmarks_weight $advanced_options_cond.landmarks_weight
-        --landmarks_file "$advanced_options_cond.landmarks_file"
-    #else if $advanced_options_cond.advanced_options == 'affine':
-        --target_affine_file "$advanced_options_cond.target_affine_file"
-        --source_affine_file "$advanced_options_cond.source_affine_file"
-    #else if $advanced_options_cond.advanced_options == 'mono':
-        --mono "true"
-    #else if $advanced_options_cond.advanced_options == 'save_transformation':
-        --target_trans_out "$target_trans_out"
-        --source_trans_out "$source_trans_out"
+    <expand macro="fiji_requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+#import os
+#import tempfile
+#set error_log = 'output_log.txt'
+#set tmp_dir = $tempfile.mkdtemp(prefix='tmp-imagej-')
+
+## ImageJ2 requires file extensions to be valid image data types.
+#set source_out_suffix = '.' + str($source_out_datatype)
+#set target_out_suffix = '.' + str($target_out_datatype)
+#set tiff_suffix = '.tiff'
+#set source_image_sans_ext = $os.path.splitext($os.path.basename($source_image.file_name))[0]
+#set source_image_with_ext = '.'.join([$source_image_sans_ext, $source_image.ext])
+ln -s '$source_image.file_name' '$source_image_with_ext' &&
+#set target_image_sans_ext = $os.path.splitext($os.path.basename($target_image.file_name))[0]
+#set target_image_with_ext = '.'.join([$target_image_sans_ext, $target_image.ext])
+ln -s '$target_image.file_name' '$target_image_with_ext' &&
+
+#set save_transformation = str($advanced_options_cond.advanced_options) == 'save_transformation'
+#set mono = str($advanced_options_cond.advanced_options) == 'mono'
+#set landmarks = str($advanced_options_cond.advanced_options) == 'landmarks'
+#set affine = str($advanced_options_cond.advanced_options) == 'affine'
+
+#if str($align_with_mask_cond.align_with_mask) == 'yes':
+    #set source_mask_sans_ext = $os.path.splitext($os.path.basename($align_with_mask_cond.source_mask.file_name))[0]
+    #set source_mask_with_ext = '.'.join([$source_mask_sans_ext, $source_mask.ext])
+    #set source_mask = $source_mask_with_ext
+    ln -s '$align_with_mask_cond.source_mask.file_name' '$source_mask_with_ext' &&
+    #set target_mask_sans_ext = $os.path.splitext($os.path.basename($align_with_mask_cond.target_mask.file_name))[0]
+    #set target_mask_with_ext = '.'.join([$target_mask_sans_ext, $target_mask.ext])
+    #set target_mask = $target_mask_with_ext
+    ln -s '$align_with_mask_cond.target_mask.file_name' '$target_mask_with_ext' &&
+#else:
+    #set source_mask = 'NULL'
+    #set target_mask = 'NULL'
+#end if
+
+## Imagej2 requires full paths to file names as well as valid image
+## data types in command line parameters, so we'll use tempfile since
+## it accomodates both of these requirements.  ImageJ2 also requires
+## files to exist before use, so each file is touched.
+#if not $mono:
+    #set tup = tempfile.mkstemp(suffix=$tiff_suffix, dir=$tmp_dir)
```

Comment:
@gregvonkuster why can you not simply create a tempdir in CWD?

```
mkdir ./temp/ &&
touch ./temp/foo.tiff &&
```

Reply:
@bgruening Java is so finicky with files on disk.  I'm not a Java expert, so I'm sure that there are better ways to deal with these file paths.  If I use relative paths to a file on disk, Java will intermittently prepend a "/" to the file name, assuming it is in the root directory on the file system.  This is one of the things I've just discovered is happening with the imagej2_bunwarpj_align tool.  This line https://github.com/bgruening/galaxytools/pull/1022/files#diff-2b1808eb0a3181c26d9399b8266e2680R19, which I use in most of these tools, will intermittently result in this exception - notice that Java has prepended a "/" to the file name which is located in the current working directory.
```
/dataset_503.tiff (No such file or directory)
Exception in thread "main" java.lang.NullPointerException
	at bunwarpj.bUnwarpJ_.alignImagesCommandLine(bUnwarpJ_.java:1252)
	at bunwarpj.bUnwarpJ_.main(bUnwarpJ_.java:1092)
```
I've found that using temple to handle these temporary disk file (ImageJ requires image files to have an appropriate extension like gif, png, etc) woks for Java because it returns an absolute path to the temporary file.  

What I haven't figured out is why Java sometimes fails with relative files, but other times works just fine.

I'm involved with some other items besides these tools this morning, but I'll be working on them throughout the day.